cronjobs/add_authority_external_ref.py

```diff
@@ -0,0 +1,178 @@
+            # The config file is not exist, assume it needs to create gnd_wiki file
+            UpdateConfigFile(config_file, "0")
+            return current_file_date_int
+    # 0 means cannot find the file on the web or there is no new file compare to last update
+    return 0
+
+
+def Main():
+    share_folder = "/mnt/ZE020150/FID-Entwicklung/ub_tools/"
+    input_file_name_for_add_auth = "input_file_for_add_authority_external_ref.txt"
+    gnd_wiki_file = "gnd_to_wiki.csv"
+
+    if len(sys.argv) != 2:
+        util.SendEmail(os.path.basename(sys.argv[0]),
+                       "This script needs to be called with an email address and the system type!\n", priority=1)
+        sys.exit(-1)
+    util.default_email_recipient = sys.argv[1]
+
+    # 1. Check whether the date of file on the web is newer compare to the last date successful update
+    print("Process 1/7 -- Check whether the file on the web is newer")
+    current_file_date_int = GetTheNewFileDate()
+    if current_file_date_int != 0:
+        # a. Download the newer file
+        newer_gz_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_int) + ".jsonld.gz"
+        newer_file_name = "authorities-gnd-person_lds_" + \
+            str(current_file_date_int) + ".jsonld"
+        print("Process 2/7 -- Downloading file")
+        if DownloadTheFile(newer_gz_file_name):
+            # b. The download file is a zip file, it needs to be extracted first
+            print("Process 3/7 -- Extracting the new file (.gz)")
+            util.ExecOrDie(util.Which("gunzip"), [
+                           "-f", f"/tmp/{newer_gz_file_name}"])
+            #  c. Get the information needed by gnd_wiki and put it into the file
+            print(
+                "Process 4/7 -- Parse the file and extract the essential information needed")
+            jq_prog_with_pipe = f"jq -c --stream '.' < /tmp/{newer_file_name} | grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > /tmp/{input_file_name_for_add_auth}"
+            if os.system(jq_prog_with_pipe) == 0:
+                # d. Run "add_authority_external_ref" program
+                print("Process 5/7 -- Generate a new gnd_wiki_file")
+                util.ExecOrDie(util.Which("add_authority_external_ref"), [
+                               "--create_mapping_file", f"/tmp/{input_file_name_for_add_auth}", f"/tmp/{gnd_wiki_file}"])
+
+                print("Process 6/7 -- Copy the gnd_wiki_file to its destination")
+                util.ExecOrDie(util.Which("cp"), [
+                               "-f", f"/tmp/{gnd_wiki_file}", f"{share_folder}{gnd_wiki_file}"])
+
+                #  e. Update the latest version date on the config file (the second line)
+                print(
+                    "Process 7/7 -- Updating the config file and remove temporary files")
+                UpdateConfigFile(
+                    config_file, current_file_date_int)
+                util.ExecOrDie(util.Which("rm"), [
```

Comment:
Maybe in these cases it would be easier to use os.remove() instead of calling an external program (rm) to remove the file